vspec/model/constants.py

```
@@ -41,96 +71,100 @@ def values(cls: Type[T]) -> Sequence[str]:
+    UPPER_CASE = VSSConstant('upper case', 'uppercase')
+    ALPHANUM_CASE = VSSConstant('alphanum case', 'alphanumcase')
+
+
+class VSSType(metaclass=VSSEnumMeta):
+    BRANCH = VSSConstant('branch', 'branch')
+    RBRANCH = VSSConstant('rbranch', 'rbranch')
+    ATTRIBUTE = VSSConstant('attribute', 'attribute')
+    SENSOR = VSSConstant('sensor', 'sensor')
+    ACTUATOR = VSSConstant('actuator', 'actuator')
+    ELEMENT = VSSConstant('element', 'element')
+
+
+class VSSDataType(metaclass=VSSEnumMeta):
+    INT8 = VSSConstant('int8', 'int8')
+    UINT8 = VSSConstant('uint8', 'uint8')
+    INT16 = VSSConstant('int16', 'int16')
+    UINT16 = VSSConstant('uint16', 'uint16')
+    INT32 = VSSConstant('int32', 'int32')
+    UINT32 = VSSConstant('uint32', 'uint32')
+    INT64 = VSSConstant('int64', 'int64')
+    UINT64 = VSSConstant('uint64', 'uint64')
+    BOOLEAN = VSSConstant('boolean', 'boolean')
+    FLOAT = VSSConstant('float', 'float')
+    DOUBLE = VSSConstant('double', 'double')
+    STRING = VSSConstant('string', 'string')
+    UNIX_TIMESTAMP = VSSConstant('UNIX Timestamp', 'UNIX Timestamp')
+    INT8_ARRAY = VSSConstant('int8[]', 'int8[]')
+    UINT8_ARRAY = VSSConstant('uint8[]', 'uint8[]')
+    INT16_ARRAY = VSSConstant('int16[]', 'int16[]')
+    UINT16_ARRAY = VSSConstant('uint16[]', 'uint16[]')
+    INT32_ARRAY = VSSConstant('int32[]', 'int32[]')
+    UINT32_ARRAY = VSSConstant('uint32[]', 'uint32[]')
+    INT64_ARRAY = VSSConstant('int64[]', 'int64[]')
+    UINT64_ARRAY = VSSConstant('uint64[]', 'uint64[]')
+    BOOLEAN_ARRAY = VSSConstant('boolean[]', 'boolean[]')
+    FLOAT_ARRAY = VSSConstant('float[]', 'float[]')
+    DOUBLE_ARRAY = VSSConstant('double[]', 'double[]')
+    STRING_ARRAY = VSSConstant('string[]', 'string[]')
+    UNIX_TIMESTAMP_ARRAY = VSSConstant('UNIX Timestamp[]', 'UNIX Timestamp[]')
+
+
+def dict_to_constant_config(name: str, info: Dict) -> Tuple[str, VSSConstant]:
+    assert ('label' in info)  # label is mandatory
```

I'd say "just access", it will throw a more intuitive `KeyError` if that's missing, no need to assert.

You can even save that into a variable, so it's accessed only once and the error will be clear, ex:

```py
label = info['label']
key = NON_ALPHANUMERIC_WORD.sub('', label).upper()
```

(and replace previous `label` with `key`

---

Thats correct, done